File: src/main/java/eotree/EOLicense.java
package eotree;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * EBNF representation:
 * <code>
 * { comment EOL } EOL
 * </code>
 */
public class EOLicense extends EONode {
    public List<EOComment> comments;

    public EOLicense(List<EOComment> comments) {
        this.comments = comments;
    }

    public EOLicense(EOComment... comments) {
        this(Arrays.stream(comments).collect(Collectors.toList()));
    }

    @Override
    public String generateEO(int indent) {
        return comments.stream()
                .map(comment -> comment.generateEO(indent))
                .collect(Collectors.joining("\n"));
    }
}

File: src/main/java/eotree/EOLicense.kt
package eotree

import java.util.Arrays
import java.util.stream.Collectors

/**
 * EBNF representation:
 * `
 * { comment EOL } EOL
` *
 */
class EOLicense(var comments: List<EOComment>) : EONode() {
    constructor(vararg comments: EOComment) : this(
        Arrays.stream<EOComment>(comments)
            .collect(Collectors.toList<EOComment>())
    )

    override fun generateEO(indent: Int): String =
        comments
            .map { comment: EOComment -> comment.generateEO(indent) }
            .joinToString("\n")

    override fun toString(): String = "+ [license]"
}
